openshift/machine-config-operator
# baremetal: ipv6, switch to NM dispatcher for DNS VIP prepending

The prepend via dhclient doesn't work via ipv6, so switch to a
NetworkManager dispatcher that runs after dhclient instead as a
workaround.

**- What I did**

Reworked the prepender implementation (with help from @celebdor) to avoid using the `prepend domain-name-servers` option in dhclient.conf - it seems this option only works for ipv4[1] and the suggested alternative of `prepend dhcp6.name-servers` also doesn't seem to work when the resolv.conf is managed by NetworkManager.

Instead we configure NetworkManager to no longer manage the resolv.conf directly, and rely on a dispatcher script which injects the necessary IP on master/worker nodes to correctly reference coredns for the baremetal platform.

[1] https://bugzilla.redhat.com/show_bug.cgi?id=643890

**- How to verify it**

Deploy with ipv4 and ipv6, confirm that on the masters the DNS VIP is prepended to the resolv.conf, and on the workers the local nic IP for the controlplane network (not the DNS VIP) is configured.

Also check `sudo journalctl -b | grep prepender` to see the log output from the dispatcher scripts.

**- Description for the changelog**

For the baremetal platform management of the resolv.conf is now handled via a NetworkManager dispatcher script, so that the necessary DNS server can be prepended for both ipv4 and ipv6 environments.
